pinecone/client.go

```diff
@@ -30,13 +72,55 @@ type NewClientParams struct {
 	SourceTag  string            // optional
 }
 
+// NewClientBaseParams holds the parameters for creating a new Client with custom authentication headers.
 type NewClientBaseParams struct {
 	Headers    map[string]string
 	Host       string
 	RestClient *http.Client
 	SourceTag  string
 }
 
+// NewClient creates and initializes a new instance of Client.
```

Comment:
I know this is largely the same docstring as above, but this is how we do it in mgmt plane, so just followed that template